File: bidly-backend/src/main/java/com/bidly/bidly_backend/model/SubastaRevision.java
package com.bidly.bidly_backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "subasta_revision")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SubastaRevision {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long identificador;

    @ManyToOne(optional = false)
    @JoinColumn(name = "subasta", unique = true, nullable = false)
    private Subasta subasta;

    @Column(nullable = false)
    private Long solicitante;

    @Column(nullable = false)
    private String estado;

    @Column(name = "fechasolicitud", nullable = false)
    private LocalDateTime fechaSolicitud;

    @Column(name = "fecharevision")
    private LocalDateTime fechaRevision;

    private String observacion;
}

File: bidly-fastapi/app/schemas/subasta_revision.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class RechazarRequest(BaseModel):
    observacion: Optional[str] = None


class SubastaRevisionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    identificador: int
    subasta: Optional[int] = None
    solicitante: Optional[int] = None
    estado: Optional[str] = None
    fechasolicitud: Optional[datetime] = None
    fecharevision: Optional[datetime] = None
    observacion: Optional[str] = None
